chore: update MSRV

jxs opened the pull request.

kalabukdima committed:
feat(gossipsub): apply `max_transmit_size` to the published message (#5642)

## Description

When trying to publish a message using gossipsub's `publish` method,
it should be possible to predict whether it will fit in the limit
defined by
the `max_transmit_size` config option.
If this limit applies to the final protobuf payload, it's not possible
to know
that in advance because the size of the added fields is not fixed.

This change makes the limit apply to the passed message size instead of
the final wire size.

## Notes & open questions

This is a minor version change because it changes the meaning of the
existing config option.
However, for the existing clients the limit will only become more
permissive, so it shouldn't break anything.

## Change checklist

<!-- Please add a Changelog entry in the appropriate crates and bump the
crate versions if needed. See
<https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [x] A changelog entry has been made in the appropriate crates

---------

Co-authored-by: Darius Clark <dariusc93@users.noreply.github.com>
Co-authored-by: João Oliveira <hello@jxs.pt>

jxs committed:
chore: update MSRV

dariusc93 reviewed (approved):
I believe we would need to bump the minor version would we not? Besides that I do agree with the bump in msrv

jxs committed:
chore(ci): address clippy beta lints (#5649)

Co-authored-by: Darius Clark <dariusc93@users.noreply.github.com>

kamuik16 committed:
feat: refactor distributed-key-value-store example (#5652)

## Description

ref #4449 

Refactored distributed-key-value-store example to use `tokio` instead of
`async-std`

## Change checklist

<!-- Please add a Changelog entry in the appropriate crates and bump the
crate versions if needed. See
<https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my
feature works
- [x] A changelog entry has been made in the appropriate crates

kamuik16 committed:
chore: refactor ping tests (#5655)

## Description

ref #4449 

Refactored ping tests to use `tokio` instead of `async-std`.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my
feature works
- [x] A changelog entry has been made in the appropriate crates

hmzdot committed:
fix(websocket): don't dial `/dnsaddr` addresses (#5613)

## Description

Returns `Error::InvalidMultiaddr` when `parse_ws_dial_addr` is called
with `/dnsaddr`.

As per its specification, `/dnsaddr` domains are not meant to be
directly dialed, instead it should be appended with `_dnsaddr.` and used
for DNS lookups afterwards

Related: #5529
Fixes: #5601 

## Notes & open questions

* Is it okay to return an error, or should I perform a DNS lookup and
resolve that DNS afterwards if address has `/dnsaddr`?
* If so, how should I handle that case where DNS lookup returns multiple
multiaddrs?

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my
feature works
- [x] A changelog entry has been made in the appropriate crates

---------

Co-authored-by: Darius Clark <dariusc93@users.noreply.github.com>

AdventureSeeker987 committed:
chore: fix some comments (#5661)

## Description

<!--
Please write a summary of your changes and why you made them.
This section will appear as the commit message after merging.
Please craft it accordingly.
For a quick primer on good commit messages, check out this blog post:
https://cbea.ms/git-commit/

Please include any relevant issues in here, for example:

Related https://github.com/libp2p/rust-libp2p/issues/ABCD.
Fixes https://github.com/libp2p/rust-libp2p/issues/XYZ.
-->

## Notes & open questions

<!--
Any notes, remarks or open questions you have to make about the PR which
don't need to go into the final commit message.
-->

## Change checklist

<!-- Please add a Changelog entry in the appropriate crates and bump the
crate versions if needed. See
<https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] A changelog entry has been made in the appropriate crates

---------

Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>

kamuik16 committed:
chore: identify::Config fields private (#5663)

## Description

Closes #5660 

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my
feature works
- [x] A changelog entry has been made in the appropriate crates

wangjingcun committed:
chore(protocols): fix some typos in comment (#5665)

## Description

<!--
Please write a summary of your changes and why you made them.
This section will appear as the commit message after merging.
Please craft it accordingly.
For a quick primer on good commit messages, check out this blog post:
https://cbea.ms/git-commit/

Please include any relevant issues in here, for example:

Related https://github.com/libp2p/rust-libp2p/issues/ABCD.
Fixes https://github.com/libp2p/rust-libp2p/issues/XYZ.
-->

fix some typos in comment



## Notes & open questions

<!--
Any notes, remarks or open questions you have to make about the PR which
don't need to go into the final commit message.
-->

## Change checklist

<!-- Please add a Changelog entry in the appropriate crates and bump the
crate versions if needed. See
<https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] A changelog entry has been made in the appropriate crates

Signed-off-by: wangjingcun <wangjingcun@aliyun.com>

jxs committed:
chore(ci): fix interop tests region, and run them again on each PR (#5666)

hanabi1224 committed:
chore: deprecate `void` crate (#5676)

## Description

<!--
Please write a summary of your changes and why you made them.
This section will appear as the commit message after merging.
Please craft it accordingly.
For a quick primer on good commit messages, check out this blog post:
https://cbea.ms/git-commit/

Please include any relevant issues in here, for example:

Related https://github.com/libp2p/rust-libp2p/issues/ABCD.
Fixes https://github.com/libp2p/rust-libp2p/issues/XYZ.
-->

The `void` crate provides a `Void` type that is conceptually equivalent
to the [`never`
type(!)](https://doc.rust-lang.org/std/primitive.never.html). This PR
tries to remove `void` crate from the dependency tree by replacing
`void::Void` with
[`std::convert::Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html)
that will eventually become an alias of the `never` type(!)

> This enum has the same role as [the ! “never”
type](https://doc.rust-lang.org/std/primitive.never.html), which is
unstable in this version of Rust. When ! is stabilized, we plan to make
Infallible a type alias to it:

## Notes & open questions

<!--
Any notes, remarks or open questions you have to make about the PR which
don't need to go into the final commit message.
-->

## Change checklist

<!-- Please add a Changelog entry in the appropriate crates and bump the
crate versions if needed. See
<https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [ ] A changelog entry has been made in the appropriate crates

kamuik16 committed:
chore: replace async-std with tokio in autonat tests (#5671)

## Description

ref #4449 

Refactored `autonat` tests to use `tokio` instead of `async-std`.

hanabi1224 committed:
chore: bump crate versions and update changelogs for #5676 (#5678)

## Description

<!--
Please write a summary of your changes and why you made them.
This section will appear as the commit message after merging.
Please craft it accordingly.
For a quick primer on good commit messages, check out this blog post:
https://cbea.ms/git-commit/

Please include any relevant issues in here, for example:

Related https://github.com/libp2p/rust-libp2p/issues/ABCD.
Fixes https://github.com/libp2p/rust-libp2p/issues/XYZ.
-->

This PR bumps crate versions and add changelog entries for crates that
are changed in #5676

Question: When should a crate version bump in the current release
process? Should it be right before or right after publishing? I see most
of current crate versions are published while some are not (e.g.
libp2p-autonat@0.13.1 libp2p-gossisub@0.48.0 and libp2p-perf@0.4.0 etc.)

## Notes & open questions

<!--
Any notes, remarks or open questions you have to make about the PR which
don't need to go into the final commit message.
-->

## Change checklist

<!-- Please add a Changelog entry in the appropriate crates and bump the
crate versions if needed. See
<https://github.com/libp2p/rust-libp2p/blob/master/docs/release.md#development-between-releases>-->

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my
feature works
- [x] A changelog entry has been made in the appropriate crates

---------

Co-authored-by: João Oliveira <hello@jxs.pt>
Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>

jxs committed:
chore(ci): add a mergify batch queue for external PRs (#5668)

and remove the if conditions no longer required as we are running them
again on every PR

---------

Co-authored-by: Guillaume Michel <guillaumemichel@users.noreply.github.com>
Co-authored-by: João Oliveira <jxs>

kamuik16 committed:
chore: refactor dcutr and gossipsub tests to use tokio instead

ref #4449

Refactored dcutr and gossipsub tests to use `tokio` instead of `async-std`.

Pull-Request: #5662.

jxs committed:
chore(gossispsub): deprecate futures-ticker

to address [RUSTSEC-2024-0384 ](https://rustsec.org/advisories/RUSTSEC-2024-0384.html). Use `futures-timer` and `Delay` instead

Pull-Request: #5674.

paulhauner committed:
chore: update FUNDING.json

I've added addresses to receive Drips donations. This is required for FIL-RetroPGF-2 in the coming days, we should get this into `master` ASAP.

This is a dedicated address controlled by Sigma Prime (including @AgeManning and myself).

Pull-Request: #5685.

jxs committed:
feat(gossipsub): introduce backpressure

superseeds #4914 with some changes and improvements, namely:

- introduce a `Delay` for `Forward` and `Publish` messages, messages that take more than the configured delay to be sent are discarded
- introduce scoring and penalize slow peers
- remove control pool
- report slow peers with the number of failed messages

Pull-Request: #5595.

hanabi1224 committed:
chore(deps): upgrade `thiserror` to 2.0

Changes:
- upgrade `thiserror` crate from `1` to `2`
- move `thiserror` to `workspace.dependencies`
- sort `workspace.dependencies`
- ~run `cargo update` to update `Cargo.lock`~
(Skipping changelog as `thiserror` does not present in any public APIs)

Pull-Request: #5689.

leopardracer committed:
fix: typos in documentation files

Pull-Request: #5693.

jxs committed:
fix(gossipsub): fix mesh/fanout inconsistencies

When a peer unsubscribes also remove it from fanout.

Pull-Request: #5690.

hanabi1224 committed:
fix(ci): Clippy Beta

Fixes CI failure in Clippy (Beta)

https://github.com/libp2p/rust-libp2p/actions/runs/12055058396/job/33614543029

Pull-Request: #5700.

jxs committed:
chore: introduce rustfmt.toml

Pull-Request: #5695.

maqi committed:
feat(kad): make Distance private field public

make Distance private field (U256) public

So that some `network density` in Distance can be calculated as below:

```rust
let density = U256::MAX / U256::from(estimated_network_size);
let density_distance = Distance(estimated_distance);
```

Pull-Request: #5705.

needsure committed:
chore: fix some typos in comment

fix some typos in comment

Pull-Request: #5721.

drHuangMHT committed:
fix(libp2p): expose builder phase error

May close #4829 and #4824.
Export three error types `BehaviourError`, `TransportError`, `WebsocketError` with rename. Feature gated `WebsocketError`.
Exported at crate root as [suggested](https://github.com/libp2p/rust-libp2p/issues/4824#issuecomment-1803013514).

Pull-Request: #5726.

bastienfaivre committed:
feat(request-response): Add connection id to behaviour events

Closes #5716.

Added connection id to the events emitted by a request-response Behaviour and adapted the code accordingly.

Pull-Request: #5719.

elenaf9 committed:
chore(ci): update Rust stable version

Update Rust stable version in our CI to the latest stable version 1.83.0.

Pull-Request: #5730.

lfgtwo committed:
chore(roadmap): fix typo

Pull-Request: #5732.

elenaf9 commented on Cargo.toml:
```diff
@@ -69,7 +69,7 @@ members = [
 
 [workspace.package]
-rust-version = "1.75.0"
+rust-version = "1.80.0"
```
```suggestion
rust-version = "1.83.0"
```

elenaf9 reviewed (commented):
Any objection for this? @jxs

Would love to be able to use `expect`, e.g. in #5242, but that requires v1.81.

hanabi1224 committed:
chore(deps): upgrade uint to 0.10

This PR upgrade `uint` to `0.10`
https://github.com/paritytech/parity-common/blob/master/uint/CHANGELOG.md#0100---2024-09-11
(Skipping changelog as there's no changes in public APIs)

Pull-Request: #5699.

hanabi1224 committed:
fix: RUSTSEC-2024-0421 by upgrading idna

https://rustsec.org/advisories/RUSTSEC-2024-0421.html

Pull-Request: #5727.

jxs committed:
deps(metrics-example): update opentelemetry to 0.27

this will help fix the `cargo deny` situation as `opentelemetry-otlp` `0.25` has `tokio` [locked to `~1.38.0`](https://crates.io/crates/opentelemetry-otlp/0.25.0/dependencies) :man_shrugging:
which then impedes us tfrom updating `netlink-sys`

Pull-Request: #5735.

jxs committed:
fix: update Cargo.lock

To finally address [RUSTSEC-2024-0384](https://rustsec.org/advisories/RUSTSEC-2024-0384).
Thanks  @hanabi1224 for submitting the [upstream update](https://github.com/rust-netlink/netlink-sys/pull/25) :heart:

Pull-Request: #5737.

jxs committed:
chore: add Unicode V3 license to deny.toml

Add `Unicode` `v3` License to our `cargo.deny` file.
This is required because of the [`icu_collections`](https://crates.io/crates/icu_collections) dependency which is authored by The Unicode Consortium.
in my ignorance, seems to be safe as the Open Source Initiative [approves it](https://opensource.org/license/unicode-license-v3), and the main `deny.toml` has also [added it](https://github.com/EmbarkStudios/cargo-deny/pull/713/files#diff-1040309c64844eb1b6b63d8fd67938adbf9461f1b3c61f12cf738f064a02d3deR50).

Pull-Request: #5738.

sergey-melnychuk committed:
chore(core): avoid unused props matching on connection.rs

Pull-Request: #5734.

thomaseizinger committed:
feat(swarm): set default for idle-connection-timeout to 10s (#4967)

## Description

With the move to a global idle-connection-timeout, connections are being
closed much more aggressively. This causes problems in situations where,
e.g. an application wants to use a connection shortly after an event has
been emitted from the `Swarm`. With a default of 0 seconds, such a
connection is instantly considered idle and therefore closed, despite
the application wanting to use it again just moments later. Whilst it is
possible to structure application code to mitigate this, it is
unnecessarily complicated.

Additionally, connections being closed instantly if not in use is a
foot-gun for newcomers to the library.

From a technical point-of-view, instantly closing idle connections is
nice. In reality, it is an impractical default. Hence, we change this
default to 10s.

10 seconds is considered to be an acceptable default as it strikes a
balance between allowing some pause between network activity, yet frees
up resources that are (supposedly) no longer needed.

Resolves: #4912.

dependabot[bot] committed:
chore(deps): bump golang.org/x/crypto from 0.21.0 to 0.31.0 (#5736)

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from
0.21.0 to 0.31.0. for the wasm webtransport tests

jxs committed:
Update Cargo.toml

Co-authored-by: Elena Frank <elena.frank@proton.me>

jameshiew committed:
fix(identify): validate public key from remote peer

Related https://github.com/libp2p/rust-libp2p/issues/5706

Discard `Info` messages received from a remote peer that contain a public key that doesn't match their peer ID, and log a warning. Don't emit `identify::Received` events to the swarm containing whatever public key they sent.

Pull-Request: #5707.

jxs committed:
deps(memory-connection-limits): update sysinfo to 0.33

Pull-Request: #5739.

stormshield-frb committed:
feat(SwarmBuilder): add with_connection_timeout method

Small PR to be able to change the `connection_timeout` value given to the `TransportTimeout` when building the `Swarm` with a `SwarmBuilder`.

Pull-Request: #5575.

xep-c-6yrpa committed:
chore(core): use `matches!` in connection.rs

Pull-Request: #5740.

elenaf9 committed:
chore: fix format with nightly

Fix formatting with nightly.

Without nightly, some of our `rustfmt.toml` rules aren't applied because they are not stable yet (e.g. our `group_imports` rule).
Our CI didn't catch it yet because it doesn't check fmt on nightly. Will fix that in a follow-up PR.

Pull-Request: #5742.

Pricstas committed:
chore(docs): fix typos in documentation

Pull-Request: #5744.

elenaf9 committed:
chore(ci): check rustfmt with nightly toolchain

Some of our rustfmt rules aren't stable yet. Thus our rustfmt CI check should run on nightly.

Pull-Request: #5743.

elenaf9 reviewed (changes requested):
While reviewing #4692 and #5266 I noticed that they also changed the rust version in the Dockerfiles, so I think we need to do this here as well?

stormshield-frb committed:
fix(kad): improve memory allocation when iterating over kbuckets

Proposal to fix https://github.com/libp2p/rust-libp2p/issues/5712.

I have changed to `ClosestIter` structure to only allocate when `kbucket_size` is higher than `K_VALUE` and only once along the life of `ClosestIter`. I think I did not break anything but I would really like some experienced people with Kademlia to take a look (@guillaumemichel :wink:).

Pull-Request: #5715.

argentpapa committed:
chore(spellchecker): fix typos in comment

- use spellchecker, and check false positive manually

```
codespell  -L crate,COMIT,comit  -S CHANGELOG.md
```
- maybe we can add typo checker in github action ref: https://github.com/codespell-project/actions-codespell

Pull-Request: #5750.

kalabukdima committed:
chore: add SQD Network to notable users list

SQD Network uses Libp2p and gossipsub protocol in particular to organize a decentralized network of p2p workers that provide fast access to stored data.

Pull-Request: #5749.

hopinheimer committed:
feat(mdns): emit `ToSwarm::NewExternalAddrOfPeer` on discovery

fixes #5104 and superseeds #5179

Pull-Request: #5753.

crStiv committed:
fix: multiple typos of different importance

This pull request addresses several typos and minor grammatical inconsistencies found across multiple files in the repository. The fixes aim to improve clarity, readability, and maintain consistency in the project's documentation and workflows.

### Key Changes:
- **`.github/ISSUE_TEMPLATE/bug_report.yml`**:
- Fixed grammatical issues in descriptions and labels.
- **`.github/ISSUE_TEMPLATE/enhancement.yml`**:
- Corrected label text and standardized phrasing.
- **`.github/ISSUE_TEMPLATE/feature_request.yml`**:
- Improved label consistency and readability.
- **`.github/pull_request_template.md`**:
- Enhanced phrasing and corrected minor grammatical errors.
- **`.github/workflows/ci.yml`**:
- Adjusted text to ensure consistent phrasing and clarity in workflow steps.

These updates ensure a more professional and coherent presentation across the project documentation and workflows.

Pull-Request: #5756.

hanabi1224 committed:
chore(deps): remove unused deps

Remove unused dependencies

(Not sure how to bump `libp2p-identity` and `libp2p-swarm-derive` versions in a PR)

Pull-Request: #5747.

elenaf9 committed:
chore(mdns): revert version bump

Revert version bump of #5753, because libp2p-mdns-0.46.1 isn't released yet.

Pull-Request: #5762.

kalabukdima committed:
deps(quic): update quinn to 0.11.6

Among other changes, it includes a fix for this issue which often reproduces with libp2p: https://github.com/quinn-rs/quinn/issues/1889

Pull-Request: #5757.

kamuik16 committed:
chore: introduce `libp2p-test-utils`

Fixes #4992

Based on the conversation in #4992.

Pull-Request: #5725.

hopinheimer committed:
feat(gossipsub): implement gossipsub 1.2 beta

This PR implements gossipsub 1.2 beta bringing changes over from lighthouse
ref PR: https://github.com/sigp/lighthouse/pull/5422

Please include any relevant issues in here, for example:
https://github.com/libp2p/specs/pull/548

Pull-Request: #5697.

hanabi1224 committed:
chore(deps): update Cargo.lock

- run `cargo update`
- lock `webrtc-ice = "=0.10.0"` to not break webrtc smoke tests
- fix `cargo clippy` warnings
- update `deny.toml` accordingly

Pull-Request: #5755.

hopinheimer committed:
feat(gossipsub): Allow setting a size threshold for IDONTWANT messages

This PR adds configurable parameter that sets minimum message size for which `IDONTWANT` messages would be send. This is an optimisation trick, discussion regarding the same can be found [here](https://github.com/sigp/lighthouse/issues/6437)

Pull-Request: #5770.

elenaf9 committed:
chore(kad): revert version bump

Revert version bump, because `libp2p-kad-0.47.0` isn't released yet.

Pull-Request: #5776.

elenaf9 committed:
chore(autonat): revert version bump

Revert version bump, `libp2p-autonat-v0.13.1` isn't released yet.

Pull-Request: #5777.

elenaf9 committed:
chore(server): revert version bump

Revert version bump, `libp2p-server-v0.12.6` isn't released yet.

Pull-Request: #5780.

elenaf9 committed:
chore(identify): revert version bump

Revert version bump, `libp2p-identify-v0.46.0` isn't released yet.

Pull-Request: #5778.

jxs committed:
chore(ci): add Zlib to deny.toml

Similar to https://github.com/libp2p/rust-libp2p/pull/5738,  [`foldhash`](https://crates.io/crates/foldhash) is the dependency that requires it. It was introduced by `hashbrown` which is dependency of a lot of crates [here](https://github.com/rust-lang/hashbrown/pull/563).
`hashbrown` is MIT, and Zlib is also an Open Source Initiative [approved license](https://opensource.org/license/zlib).

If you prefer we can also do as the upstream `cargo-deny` do and just add `foldhash` [to the exceptions](https://github.com/EmbarkStudios/cargo-deny/pull/618/files#diff-1040309c64844eb1b6b63d8fd67938adbf9461f1b3c61f12cf738f064a02d3deR56) but I can't see any advantage to it.
Cc @hanabi1224

Pull-Request: #5769.

elenaf9 committed:
chore(allow-block-list): revert version bump

Revert version bump, `libp2p-allow-block-list-v0.4.1` isn't released yet.

Pull-Request: #5779.

drHuangMHT committed:
chore(kad): remove default constructor for ProtocolConfig

Remove items that were deprecated in #5122.

Pull-Request: #5774.

hopinheimer committed:
feat: broadcasting `idontwant` for published messages

This PR introduces an optimisation to preemptively broadcast `IDONTWANT` for published messages, preventing redundant downloads of the same message received over gossip. This feature is opt-in and respects the existing `idontwant_message_size_threshold`. By default, `IDONTWANT` messages will only be sent for published messages larger than 1000 bytes.

reference PRs:
[spec](https://github.com/libp2p/specs/pull/642)

Pull-Request: #5773.

jxs committed:
update Dockerfiles Rust version

jxs committed:
update CHANGELOG.md

jxs committed:
fixup! update CHANGELOG.md and Cargo.toml

jxs committed:
Merge branch 'master' of github.com:libp2p/rust-libp2p into update-msrv

jxs committed:
update main Cargo.toml

jxs committed:
update main Cargo.lock

elenaf9 reviewed (approved):
I think something went wrong in the rebase? PR suddenly has 71 commits.

But rest LGTM!

jxs committed:
set the runner container to be bullseye

jxs committed:
Merge branch 'master' of github.com:libp2p/rust-libp2p into update-msrv
